flutter/lib/src/event_processor/android_platform_exception_event_processor.dart

```diff
@@ -103,34 +103,29 @@ class AndroidPlatformExceptionEventProcessor implements EventProcessor {
           .toList(growable: true);
       jvmThreads.add(first);
     }
-
-    return event.copyWith(
-      exceptions: [
-        ...?event.exceptions,
-        ...jvmExceptions,
-      ],
-      threads: [
-        ...?threads,
-        if (_options.attachThreads) ...jvmThreads,
-      ],
-    );
+    event.exceptions = [
+      ...?event.exceptions,
+      ...jvmExceptions,
+    ];
+    event.threads = [
+      ...?threads,
+      if (_options.attachThreads) ...jvmThreads,
+    ];
+    return event;
   }
 
   /// If the crash originated on Android, the Dart side didn't crash.
   /// Mark it accordingly.
   List<SentryThread>? _markDartThreadsAsNonCrashed(
     List<SentryThread>? threads,
   ) {
-    return threads
-        ?.map(
-          (e) => e.copyWith(
-            crashed: false,
-            // Isolate is safe to use directly,
-            // because Android is only run in the dart:io context.
-            current: e.name == Isolate.current.debugName,
-          ),
-        )
-        .toList(growable: false);
+    return threads?.map((thread) {
```

Comment:
Isn't it enough now to iterate over each element and mutate it? You don't actually need to create a new list